projects/core/src/anonymous-consents/store/effects/anonymous-consents.effect.ts

```diff
@@ -99,31 +97,98 @@ export class AnonymousConsentsEffects {
     ),
     filter(([, registerAction]) => Boolean(registerAction)),
     concatMap(() =>
-      this.anonymousConsentService
-        .getAnonymousConsentTemplate(
-          this.anonymousConsentsConfig.anonymousConsents.registerConsent
-        )
-        .pipe(
-          withLatestFrom(
-            this.anonymousConsentService.getAnonymousConsent(
-              this.anonymousConsentsConfig.anonymousConsents.registerConsent
-            ),
-            this.authService.getOccUserId()
-          ),
-          map(([template, consent, userId]) => {
+      this.anonymousConsentService.getAnonymousConsents().pipe(
+        withLatestFrom(
+          this.authService.getOccUserId(),
+          this.anonymousConsentService.getAnonymousConsentTemplates(),
+          this.authService.isUserLoggedIn()
+        ),
+        filter(([, , , loggedIn]) => loggedIn),
+        concatMap(([consents, userId, templates]) => {
+          const actions: UserActions.TransferAnonymousConsent[] = [];
+          for (const consent of consents) {
             if (
               consent.consentState ===
-              ANONYMOUS_CONSENT_STATUS.ANONYMOUS_CONSENT_GIVEN
+                ANONYMOUS_CONSENT_STATUS.ANONYMOUS_CONSENT_GIVEN &&
+              (!this.anonymousConsentsConfig.anonymousConsents
+                .requiredConsents ||
+                !this.anonymousConsentsConfig.anonymousConsents.requiredConsents.includes(
+                  consent.templateCode
+                ))
             ) {
-              return new UserActions.TransferAnonymousConsent({
-                userId: userId,
-                consentTemplateId: template.id,
-                consentTemplateVersion: template.version,
-              });
+              for (const template of templates) {
+                if (template.id === consent.templateCode) {
+                  actions.push(
+                    new UserActions.TransferAnonymousConsent({
+                      userId,
+                      consentTemplateId: template.id,
+                      consentTemplateVersion: template.version,
+                    })
+                  );
+                  break;
+                }
+              }
             }
-            return EMPTY;
-          })
-        )
+          }
+          if (actions.length > 0) {
+            return actions;
+          }
+          return EMPTY;
+        })
+      )
+    )
+  );
+
+  @Effect()
+  giveMandatoryConsentsToUser$: Observable<
```

Comment:
What's the reason that this is a separate effect? Can this logic be done in `transferAnonymousConsentsToUser$`? (in which case the effect should be rename ofc)

Reply:
I thought about it. I do think it would make for a one even more complicated effect or maybe not. Since the first one is only after register I like having it separate, because the other should happen after every login.